sklearn/model_selection/tests/test_split.py

```diff
@@ -0,0 +1,723 @@
+        self.dummy_str = dummy_str
+        self.dummy_obj = dummy_obj
+        if callback is not None:
+            callback(self)
+
+        if self.allow_nd:
+            X = X.reshape(len(X), -1)
+        if X.ndim >= 3 and not self.allow_nd:
+            raise ValueError('X cannot be d')
+        if sample_weight is not None:
+            assert_true(sample_weight.shape[0] == X.shape[0],
+                        'MockClassifier extra fit_param sample_weight.shape[0]'
+                        ' is {0}, should be {1}'.format(sample_weight.shape[0],
+                                                        X.shape[0]))
+        if class_prior is not None:
+            assert_true(class_prior.shape[0] == len(np.unique(y)),
+                        'MockClassifier extra fit_param class_prior.shape[0]'
+                        ' is {0}, should be {1}'.format(class_prior.shape[0],
+                                                        len(np.unique(y))))
+        if sparse_sample_weight is not None:
+            fmt = ('MockClassifier extra fit_param sparse_sample_weight'
+                   '.shape[0] is {0}, should be {1}')
+            assert_true(sparse_sample_weight.shape[0] == X.shape[0],
+                        fmt.format(sparse_sample_weight.shape[0], X.shape[0]))
+        if sparse_param is not None:
+            fmt = ('MockClassifier extra fit_param sparse_param.shape '
+                   'is ({0}, {1}), should be ({2}, {3})')
+            assert_true(sparse_param.shape == P_sparse.shape,
+                        fmt.format(sparse_param.shape[0],
+                                   sparse_param.shape[1],
+                                   P_sparse.shape[0], P_sparse.shape[1]))
+        return self
+
+    def predict(self, T):
+        if self.allow_nd:
+            T = T.reshape(len(T), -1)
+        return T[:, 0]
+
+    def score(self, X=None, Y=None):
+        return 1. / (1 + np.abs(self.a))
+
+    def get_params(self, deep=False):
+        return {'a': self.a, 'allow_nd': self.allow_nd}
+
+
+X = np.ones((10, 2))
+X_sparse = coo_matrix(X)
+W_sparse = coo_matrix((np.array([1]), (np.array([1]), np.array([0]))),
+                      shape=(10, 1))
+P_sparse = coo_matrix(np.eye(5))
+y = np.arange(10) // 2
+
+
+@ignore_warnings
+def test_cross_val_generator_with_default_indices():
+    X = np.array([[1, 2], [3, 4], [5, 6], [7, 8]])
+    y = np.array([1, 1, 2, 2])
+    labels = np.array([1, 2, 3, 4])
+    test_folds = np.array([1, 1, 2, 2])
+    loo = LeaveOneOut()
+    lpo = LeavePOut(2)
+    kf = KFold(2)
+    skf = StratifiedKFold(2)
+    lolo = LeaveOneLabelOut()
+    lopo = LeavePLabelOut(2)
+    ss = ShuffleSplit()
+    ps = PredefinedSplit()
+    for cv in [loo, lpo, kf, skf, lolo, lopo, ss, ps]:
+        print cv
+        for train, test in cv.split(X, y, labels):
+            assert_not_equal(np.asarray(train).dtype.kind, 'b')
+            assert_not_equal(np.asarray(train).dtype.kind, 'b')
+            X[train], X[test]
```

Comment:
What does this line check (I know it was there before but I don't didn't understand it)? The indexing? maybe we want to check that the sizes add up to ``n_samples``? Or is that not required? [I guess not for predefined split]